deps/chakrashim/src/v8debug.cc

```diff
@@ -21,71 +21,715 @@
 #include "v8.h"
 #include "v8-debug.h"
 #include "jsrtutils.h"
+#include "chakra_natives.h"
 
 namespace v8 {
 
-__declspec(thread) bool g_EnableDebug = false;
-static JsContextRef g_debugContext = JS_INVALID_REFERENCE;
+  __declspec(thread) bool g_EnableDebug = false;
 
-bool Debug::EnableAgent(const char *name, int port, bool wait_for_connection) {
-  HRESULT hr = S_OK;
+  MessageQueue Debug::messageQueue;
+  Debug::MessageHandler Debug::handler = nullptr;
 
-#ifndef NODE_ENGINE_CHAKRACORE
-  if (!g_EnableDebug) {
-    // JsStartDebugging needs COM initialization
-    IfComFailError(CoInitializeEx(nullptr, COINIT_MULTITHREADED));
+  MessageQueue::MessageQueue() {
+    this->waitingForMessage = false;
+    this->isProcessingDebuggerMsg = false;
+    this->debugEventProcessCount = 0;
+    this->msgQueue = new std::queue<wchar_t*>();
 
+    int err = uv_sem_init(&newMsgSem, 0);
+    CHAKRA_VERIFY(err == 0);
+    err = uv_mutex_init(&msgMutex);
+    CHAKRA_VERIFY(err == 0);
+  }
+
+  MessageQueue::~MessageQueue() {
+    uv_mutex_destroy(&msgMutex);
+    uv_sem_destroy(&newMsgSem);
+    delete this->msgQueue;
+    this->msgQueue = nullptr;
+  }
+
+  void MessageQueue::SaveMessage(const uint16_t* command, int length) {
```

Comment:
On windows uint16_t and wchar_t are both 2 bytes and debugging support is currently only for windows. I have a static_assert to make sure we fix this for xplat